test/browsercontext-storage-state.spec.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2018 Google Inc. All rights reserved.
+ * Modifications copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { it, expect } from './fixtures';
+
+it('should capture local storage', async ({ browser }) => {
+  const context1 = await browser.newContext();
+  const page1 = await context1.newPage();
+  await page1.route('**/*', route => {
+    route.fulfill({ body: '<html></html>' }).catch(() => {});
+  });
+  await page1.goto('https://www.example.com');
+  await page1.evaluate(() => {
+    localStorage['name1'] = 'value1';
+  });
+  const { origins } = await context1.storageState();
```

Comment:
Let's navigate to another origin, to check that we handle multiple origins correctly.